extensions/README.md

```diff
@@ -47,22 +47,48 @@ types. This is done via three distinct extension mechanisms:
+a corresponding `extensions.Point`. These are:
+
+| type                  | extension point                                               |
+| --------------------- | ------------------------------------------------------------- |
+| `comid.Comid`         | `comid.ExtComid`                                              |
+| `comid.Entity`        | `comid.ExtEntity`                                             |
+| `comid.FlagsMap`      | `comid.ExtReferenceValueFlags`, `comid.ExtEndorsedValueFlags` |
+| `comid.Mval`          | `comid.ExtReferenceValue`, `comid.ExtEndorsedValue`           |
+| `comid.Triples`       | `comid.ExtTriples`                                            |
+| `corim.Entity`        | `corim.ExtEntity`                                             |
+| `corim.Signer`        | `corim.ExtSigner`                                             |
+| `corim.UnsignedCorim` | `corim.ExtUnsignedCorim`                                      |
+
+Note that `comid.Mval` and `comid.FlagsMap` are used for both reference and
+endorsed values, which may be extended separately. This is why there are two
+extension points associated with each. Additionally, `comid.ExtMval` and
+`comid.ExtFlags` also exist for when you want to register extensions with a
+`comid.Mval` or `comid.Measurment` (and so don't have the context of whether it
+will be going into a reference or an endorsed value).
+
+The diagram below shows a visual representation of where these extension points
+originate in the `struct` hierarchy, and which CoRIM object are "aware" of
+which extension points:
+
+![map extensions](corim-map-extensions.png?)
+
+(note: the diagram shows the logical relationships between structures and does
+not meant to accurately reflect the code -- e.g. it omits the container types,
+representing them as slices)
 
 To extend the above types, you need to define a struct containing your
-extensions and pass a pointer to an instance of that struct to the
-`RegisterExtensions()` method of the corresponding instance of the type that is
-being extended. This should be done as early as possible, before any marshaling
-is performed.
+extensions for each extension point that you want to extend. You then need to
+create an `extensions.Map` that maps the extension points  to a pointer to an
+instance of the corresponding struct. Finally, you need to pass  the map to the
+`RegisterExtensions()` method of an instance of the top-most type that is being
+extended. For example, if you want to extend the comid, reference values, and
+entities, you would create a struct defining the extensions for each, call
+`extensions.NewMap()` to create a new map, call `Add()` on the map to add
+mappings from the three extension points to pointers to empty instances of your
+structs, and, finally, pass the map to `comid.Comid.RegisterExtensions()`. This
+should be done as early as possible, before any marshaling is performed.
```

Comment:
Should you be referring to the snippet here which one should refer as an example?